golang/cosmos/x/swingset/types/msgs.go

```diff
@@ -20,16 +20,28 @@ func NewMsgDeliverInbound(msgs *Messages, submitter sdk.AccAddress) *MsgDeliverI
 	}
 }
 
-func (msg MsgDeliverInbound) CheckAdmissibility(ctx sdk.Context, callToController func(sdk.Context, string) (string, error)) error {
-	// TODO: This is where we would consult the controller for deterministic
-	// advice on whether we are overwhelmed.
+func (msg MsgDeliverInbound) CheckAdmissibility(ctx sdk.Context, data interface{}) error {
+	keeper, ok := data.(SwingSetKeeper)
+	if !ok {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "data must be a SwingSetKeeper, not a %T", data)
+	}
+
 	/*
 		// The nondeterministic torture test.  Mark every third message as not inadmissible.
 		if rand.Intn(3) == 0 {
 			return fmt.Errorf("FIGME: MsgDeliverInbound is randomly not admissible")
 		}
 	*/
-	return nil
+
+	// Charge the submitter for the message parameters.
+	beansPerUnit := keeper.GetBeansPerUnit(ctx)
+	beans := beansPerUnit[BeansPerInboundTx]
```

Comment:
Just checking.. is `CheckAdmissibility` guaranteed to be called once per txn, with *all* the swingset-bound messages that txn contains? Or is it called separately for each swingset-bound message found in a txn?

Reply:
It's called separately for each swingset-bound "DELIVER_INBOUND" message found in a Cosmos txn.